plugins/secretstores/docker/docker.go

```diff
@@ -0,0 +1,83 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package docker
+
+import (
+	_ "embed"
+	"errors"
+	"os"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/secretstores"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+// mount directory created by Docker when using Docker Secrets
+const dockerSecretsDir string = "/run/secrets"
+
+type Docker struct {
+	ID string `toml:"id"`
+}
+
+func (*Docker) SampleConfig() string {
+	return sampleConfig
+}
+
+// Init initializes all internals of the secret-store
+func (d *Docker) Init() error {
+	if d.ID == "" {
+		return errors.New("id missing")
+	}
+	if _, err := os.Stat(dockerSecretsDir); os.IsNotExist(err) {
+		return errors.New("/run/secrets directory does not exist")
+	}
+	return nil
+}
+
+func (d *Docker) Get(key string) ([]byte, error) {
+	secretFile := dockerSecretsDir + "/" + key
+	value, err := os.ReadFile(secretFile)
+	if err != nil {
+		return nil, errors.New("cannot find the secret file under /run/secrets")
+	}
+	if string(value) == "" {
+		return nil, errors.New("the value of the secrets file is empty")
+	}
+	return value, nil
+}
+
+func (d *Docker) List() ([]string, error) {
+	secrets := make([]string, 0)
+	secretFiles, err := os.ReadDir(dockerSecretsDir)
+	if err != nil {
+		return nil, errors.New("cannot read files under /run/secrets directory")
+	}
+	if len(secretFiles) == 0 {
+		return nil, errors.New("cannot find any secrets under /run/secrets")
+	}
+	for _, entry := range secretFiles {
+		secrets = append(secrets, entry.Name())
+	}
+	return secrets, nil
+}
+
+func (d *Docker) Set(_, _ string) error {
+	return errors.New("secret-store does not support creating secrets")
+}
+
+// GetResolver returns a function to resolve the given key.
+func (d *Docker) GetResolver(key string) (telegraf.ResolveFunc, error) {
+	resolver := func() ([]byte, bool, error) {
+		s, err := d.Get(key)
+		return s, false, err
```

Comment:
Does it make sense to support also dynamically changing secrets?  I.e. to support cases where the secret might be changed during runtime...

Reply:
AFAIK and have confirmed with the team working with Docker Compose, the secrets mounted within containers are read-only mounted file systems. So values cannot be altered within the container or at runtime. 

From one of the developers for Compose v2

> ro in container, but if you change it on host it will be modified. compose "secrets" are just bind mounts, as a "development mode" for swarm secrets (by the time)

Reply:
@shantanoo-desai that's what I meant. I'm aware of the fact that you cannot modify the secrets from _within_ the container (thus setting secrets is not supported). My use-case was that you change a secret from the host (i.e. from outside the container) e.g. if a token expires or similar.

Reply:
How is this resolved? My question is: Does it make sense to reread the secrets every time the secret is accessed to allow for dynamic secrets that can be changed from the host e.g. after a token etc expired?

Reply:
I am still not very confident regarding the dynamic secrets you are mentioning. Let me perform more concrete tests and check if this is actually the case that the changes on host are reflected _dynamically_ into the container.
 
 When you mention reread the secrets, what can be the potential solution for the `GetResolver` here. I am a bit lost here.

Reply:
will be set via a `dynamic` configuration. Defaults to `false` .